blueoil/tfds_data_processor.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright 2018 The Blueoil Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+import pprint
+import tensorflow as tf
+
+
+class TFSequence:
+    """Sequence several processor together.
+
+    Args:
+        processors (List[Processor]): list of processor.
+
+    Examples:
+        | *TFSequence([*
+        |     *TFFlipLeftRight(0.5),*
+        |     *TFHue((-10, 10)),*
+        | *])*
+    """
+
+    def __init__(self, processors):
+        self.processors = processors
+
+    @tf.function
```

Comment:
`_tfds_map_func` is also decorated with `@tf.function`. Isn't it enough?

Reply:
As you said, this was not necessary...
I will remove `TFSequence` and replace with `Sequence`

Reply:
@tfujiwar 
I think that was necessary maybe because the function that uses for-loop or if-else should be directly wrapped by `@tf.function`.
Actually I failed tests when replacing `TFSequence` with `Sequence`.
But there seem less documents for tf.function in tf 1.xx, how do you think?

Reply:
@kurarrr Thanks for checking. I understood the situation.
- `tf.data.dataset.map` requires a function that can be run as a graph
- if-else blocks and for-loops cannot be run as a graph
- `@tf.function` converts if-else blocks and for-loops to tf.operators automatically

That's why we need `@tf.function` for both `_tfds_map_func` and `__call__`. So the current code is fine 👍 

https://www.tensorflow.org/tutorials/customization/performance?hl=en#autograph_conditionals

Reply:
@tfujiwar 
This document is exactly what I was looking for.  Thanks!